### PR TITLE
Decouple SideNav from default layout

### LINKED ISSUE #9427: Decouple the sidenav from the default layout to create a standalone component
In support of updating the sidenav, we should first decouple the default layout to a separate component. 

Design: https://xd.adobe.com/view/1453b6a5-ca9c-4794-8fbf-34f1349e45a3-4100/

Rely on the epic rancher/dashboard#9254 for full details but here's some helpful info from there:

----

Added by @aalves08 on 19/07/2023:

I think that this is a great opportunity to decouple the sidenav (make it as a component) from the default layout. We would benefit a lot from that work. There's similar work done here https://github.com/codyrancher/rancher-desktop/pull/1/files#diff-0c0232aeeb5da2a0b2ff8dadf8922e23b71d4c14da75d3d9aa1d99ca66937aec which can give great insights on how to achieve it with the current logic.

Added by @rak-phillip on 19/07/2023:

We also attempted to isolate the sidenav a while back but were not ready to take it. https://github.com/rancher/dashboard/pull/8000

### PR DESCRIPTION
**Pull request by aalves08**
Fixes #9427 

- Decouples `SideNav` from `default` layout

### Testing
As far as tests go, we have quite a few e2e tests that cover the usage of the `SideNav` (check vscode screenshots). Stuff like group or individual items visibility in specific scenarios are covered, as well navigating to both single and group items.

- Tested in multiple scenarios in Dashboard usage
- Tested in standalone mode (with `epinio` - Check screenshots)
- **missing testing with Harvester!!!**

### Screenshots
**_current e2e tests_**
Screenshot 2023-08-04 at 11 40 29
Screenshot 2023-08-04 at 11 40 51
Screenshot 2023-08-04 at 11 41 24
Screenshot 2023-08-04 at 11 41 31

**_epinio standalone_**
Screenshot 2023-08-04 at 12 21 10
Screenshot 2023-08-04 at 12 23 24
Screenshot 2023-08-04 at 12 23 35
Screenshot 2023-08-04 at 12 24 24

### CODE REVIEW

**Comment by aalves08** [comment]
@richard-cox doesn't this need to move forward in order to complete the objective?

File: shell/components/SideNav.vue [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+      gettingGroups: false
+    };
+  },
+
+  created() {
+    this.queueUpdate = debounce(this.getGroups, 500);
+
+    this.getGroups();
+
+    // await this.$store.dispatch('prefs/setLastVisited', this.$route);
```
Comment (richard-cox) [inline shell/components/SideNav.vue:38]: to remove

File: shell/components/SideNav.vue [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+    },
+
+    groupSelected(selected) {
+      this.$refs.groups.forEach((grp) => {
+        if (grp.canCollapse) {
+          grp.isExpanded = (grp.group.name === selected.name);
+        }
+      });
+
+      // this.$emit('expand', selected);
```
Comment (richard-cox) [inline shell/components/SideNav.vue:384]: to remove, not in default layout groupSelected

Comment (aalves08) [reply shell/components/SideNav.vue:384]: Don't know where this one came from... 🤔

**Comment by aalves08** [comment]
> * 4 `NAV` styles need moving to `SideNav`

@richard-cox I can only move 3 out of 4. One that needs to stay is regarding the grid-area definition + the styles associated with it seem to be from a global setting, which doesn't seem to fit any of the classes on `SideNav`.

**Review by richard-cox — CHANGES_REQUESTED** [review]
`NAV` is the top level `nav` element. All four can move across as is to the root of `styles`, or all 4 can move under `.side-nav` (minus `NAV`)

**Review by richard-cox — APPROVED** [review]

**Pull request merged**